backend/src/main/java/com/rota/auth/AudienceValidator.java

```diff
@@ -0,0 +1,23 @@
+package com.rota.auth;
+
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidatorResult;
+import org.springframework.security.oauth2.jwt.Jwt;
+
+class AudienceValidator implements OAuth2TokenValidator<Jwt> {
+  private final String audience;
+
+  AudienceValidator(String audience) {
+    this.audience = audience;
+  }
+
+  public OAuth2TokenValidatorResult validate(Jwt token) {
+    OAuth2Error error = new OAuth2Error("invalid_token", "The required audience is missing", null);
+
+    if (token.getAudience().contains(audience)) {
+      return OAuth2TokenValidatorResult.success();
+    }
+    return OAuth2TokenValidatorResult.failure(error);
+  }
```

Comment:
Not sure if it would make this easier to read, but we can also write this as
```
return token.getAudience().contains(audience)
  ? OAuth2TokenValidatorResult.success()
  : OAuth2TokenValidatorResult.failure(
      new OAuth2Error("invalid_token", "The required audience is missing", null));
```

Reply:
If people find that easier to read I'm happy to make this refactor.